### PR TITLE
Give better failure report on Azure Deployment Failure.

### PR DESCRIPTION
**Pull request by orsenthil**
https://jira.mesosphere.com/browse/QUALITY-1837

A `RetryError` is not very useful Exception. 

When a RetryError happens, we should give more diagnostic information and fail with a better exception like `DeploymentError`.

### CODE REVIEW

**Comment by orsenthil** [comment]
@margaret  / @cprovencher  - Please review this.

**Review by margaret — APPROVED** [review]
This looks good. I think we need a separate ticket to add some tests where the are some errors. Right now I think the mocks are only ever return good responses.

File: dcos_launch/platforms/arm.py [hunk 1]
```diff
@@ -223,28 +227,41 @@ def wait_for_deployment(self, timeout=60 * 60):
...
-                deploy_ops = self.azure_wrapper.rmc.deployment_operations.list(
-                    self.group_name, DEPLOYMENT_NAME.format(self.group_name))
-                failures = [(op.properties.status_code, op.properties.status_message) for op
-                            in deploy_ops if op.properties.provisioning_state == 'Failed']
-                for failure in failures:
-                    log.error('Deployment operation failed! {}: {}'.format(*failure))
-                raise Exception('Deployment Failed!')
+                log.info('Deployment failed. Checking deployment operations.')
+                azure_failure_report()
+                raise DeploymentError('Azure Deployment Failed!')
```
Comment (cprovencher) [inline dcos_launch/platforms/arm.py:249]: For consistency with the rest of dcos-launch I would use this error format:
https://github.com/dcos/dcos-launch/blob/master/dcos_launch/arm.py#L55

Comment (orsenthil) [reply dcos_launch/platforms/arm.py:249]: Hey @cprovencher  and @margaret  - Using `LauncherError` does not seem right to me.
It is not a **LauncherError* (like it missed ssh keys or a field was invalid). I have moved `DeploymentError` to a `util.py`.

Also, on the format of current DeploymentError. The existing format is new to me.

* It is not calling Super()

The right way to do Custom Exception is simply inherit from Exception and if you are doing anything futher, call super() for the result of the flow.

More information here 

* https://stackoverflow.com/a/1319675
* https://www.programiz.com/python-programming/user-defined-exception
* https://julien.danjou.info/python-exceptions-guide/

**Pull request merged**